src/webview/columns/data-column.tsx

```diff
@@ -31,67 +55,240 @@ export class DataColumn implements ColumnContribution {
     readonly label = 'Data';
     readonly priority = 1;
 
-    protected byteGroupStyle: React.CSSProperties = {
-        marginRight: `${DataColumn.Styles.MARGIN_RIGHT_PX}px`
+    render(range: BigIntMemoryRange, memory: Memory, options: TableRenderOptions): React.ReactNode {
+        const byteGroups = this.parse({ memory, range, renderOptions: options });
+        return byteGroups.map(group =>
+            <EditableDataColumnGroup key={group.startAddress.toString(16)} range={range} group={group} options={options} writeMemory={this.writeMemory}></EditableDataColumnGroup>);
+    }
+
+    protected writeMemory = async (writeArguments: DebugProtocol.WriteMemoryArguments): Promise<void> => {
+        await messenger.sendRequest(writeMemoryType, HOST_EXTENSION, writeArguments);
     };
 
-    render(range: BigIntMemoryRange, memory: Memory, options: TableRenderOptions): React.ReactNode {
-        return this.renderGroups(range, memory, options);
-    }
-
-    protected renderGroups(range: BigIntMemoryRange, memory: Memory, options: TableRenderOptions): React.ReactNode {
-        const groups = [];
-        let words: React.ReactNode[] = [];
-        for (let address = range.startAddress; address < range.endAddress; address++) {
-            words.push(this.renderWord(memory, options, address));
-            if (words.length % options.wordsPerGroup === 0) {
-                this.applyEndianness(words, options);
-                const isLast = address + 1n >= range.endAddress;
-                const style: React.CSSProperties | undefined = isLast ? undefined : this.byteGroupStyle;
-                groups.push(<span className='byte-group hoverable' data-column='data' style={style} key={address.toString(16)}>{words}</span>);
+    // Parse the provided memory, so that we can separate UI from data
+    protected parse(options: { range: BigIntMemoryRange, memory: Memory, renderOptions: TableRenderOptions }): DataColumnByteGroup[] {
```

Comment:
I think this function is happening too early. Every part of this column has to do with handling slices of `Memory` pool, The closer we stay to that reality, the more flexible the implementation, and the cheaper, since we spend the most time passing around pointers and wide pointers, wait as long as possible to allocate new representations of the memory. Trying to do some refactoring of this code to try out different styles of editability, I found this early divorce from the underlying data the greatest obstacle.

Reply:
Good point, initially I wanted to separate the UI part from the data, however, as you mentioned, we loose flexibility by doing that. Thank you for highlighting it!

Reply:
Would changing the behavior in this PR cause a lot of issues for you? Should I let it as it is or update it? 🤔

Reply:
I'd prefer to update this in this PR, as it will facilitate later changes. As a bonus, I think going back closer to the old style will also be slightly more performant, as we won't be creating as many new object representations of the same memory chunks. My suggestion would be to make the editable group accept as props the options, memory, row-range (we technically only need its end address), and then either a start address or a range for the group. Then the style of current `main` that builds by iterating over addresses and byte offsets can be restored.